internal/provider/resource_apply_module.go

```diff
@@ -554,6 +685,60 @@ func (t *applyModuleResource) getCurrentApplied(ctx context.Context,
 	return nil
 }
 
+// extractRunError extracts the error from a run's logs (if the run errored out).
+func (t *applyModuleResource) extractRunError(ctx context.Context, run *sdktypes.Run) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if run.Status == sdktypes.RunErrored {
+		var jobID string
+
+		switch {
+		case run.Apply != nil:
+			if run.Apply.CurrentJobID != nil {
+				jobID = *run.Apply.CurrentJobID
+			}
+		case run.Plan != nil:
+			if run.Plan.CurrentJobID != nil {
+				jobID = *run.Plan.CurrentJobID
+			}
+		}
+
+		if jobID == "" {
+			diags.AddWarning("Run status is errored, but no job ID found", "")
+			return diags
+		}
+
+		logs, err := t.client.Job.GetJobLogs(ctx, &sdktypes.GetJobLogsInput{ // unlimited
+			JobID: jobID,
+		})
+		if err != nil {
+			diags.AddError("Failed to get job logs", err.Error())
+			return diags
+		}
+
+		// Find the first mention of "error" in the logs.
+		splitLogs := strings.Split(logs.Logs, "\n")
+		foundIx := -1
+		for i, log := range splitLogs {
+			if strings.HasPrefix(strings.ToLower(log), "error") {
```

Comment:
This should not include the "Created new state version" log or anything after it.